### ISSUE TITLE
Maintain modified Rust standard libraries and `translate_libs.sh` in a separate repo

### ISSUE THREAD

**Issue by RyanGlScott** [GaloisInc/crucible#1252]
Currently, we vendor in our modified copies of the Rust standard libraries (as well as the associated `translate_libs.sh` script, which runs `mir-json` on these modified standard libraries to compile them) as a subdirectory in `crux-mir`. Historically, this made sense, as `crux-mir` was the sole consumer of these files. It is unclear if this arrangement makes as much sense nowadays, however, for the following reasons:

1. Both `crux-mir` and SAW (the latter of which now has a MIR backend) need to make use of `translate_libs.sh`. Telling SAW users to download `crux-mir` for the sole purpose of running `translate_libs.sh` doesn't make much sense, given that `crux-mir` is otherwise unrelated to SAW.
2. The vendored-in copies of the Rust standard libraries are actually quite large (on the order of several megabytes), which substantially increases the amount of time it takes to clone the `crucible` repo. This is a bit of a shame, considering that many `crucible` developers aren't going to be using the MIR-related parts.

To address both of these concerns, I propose that we factor out the modified Rust standard libraries (and `translate_libs.sh`) into a separate repo. I'll tentatively call this repo "`mir-json-libs`" until I can think of something better. That way, `mir-json-libs` development can live outside of `crucible` (thereby avoiding further bloating of the `crucible` repo), and there is now a central location for further standard library modifications needed to support `crux-mir` and SAW.

**Comment by sauclovian-g** [comment]
Won't it still need to be a submodule of crucible (or at least want to be, for the same reasons we just added mir-json in saw)? That kind of defeats point 2. Not that this matters, since point 1 is ample reason to do this :-)

**Comment by RyanGlScott** [comment]
> Won't it still need to be a submodule of crucible (or at least want to be, for the same reasons we just added mir-json in saw)? That kind of defeats point 2.

Yes, that is fair. Let me revise point (2), then: one disadvantage of versioning the `crux-mir` standard libraries in the same repo as `crucible` is that the `git` history for the `crux-mir` standard libraries is _very_ large and messy, and it would be better to keep it as separate as possible. (For an example of what I'm taking about, check out the sheer number of commits from the last time we updated these standard libraries: https://github.com/GaloisInc/mir-json/pull/49)

**Comment by sauclovian-g** [comment]
Oof. point taken 😀

**Comment by RyanGlScott** [comment]
An alternative idea: instead of moving the code for the forked standard libraries into its own repo, we could instead move them into the [`mir-json`](https://github.com/GaloisInc/mir-json) repo. This would have the following advantages:

* Crux-MIR and SAW users have to install `mir-json` anyway, so there is some logic in putting the standard libraries in the same location.
* When https://github.com/GaloisInc/mir-json/issues/67 is fixed, users will _need_ to build these modified standard libraries, so putting them next to where `mir-json` lives will make it easier to remember to do this.

A downside is that the commit history for `mir-json` will become muddied by the commit history for the standard libraries, as noted above. Given that the pace of `mir-json` development is far less rapid than the pace of development for `crucible`, my expectation is that this wouldn't be too bad in practice.

**Comment by sauclovian-g** [comment]
That seems reasonable. (Also, realistically, I'd expect sometimes there'll be changes that span both `mir-json` and the standard libraries, and having them in one repo will make those simpler.)
